aspnetcore/migration/1x-to-2x/identity-2x.md

```diff
@@ -0,0 +1,272 @@
+                o.LogoutPath = "/Account/LogOff";
+            });
+    ```
+
+### JWT Bearer Authentication
+Make the following changes in *Startup.cs*:
+
+- Remove the `UseJwtBearerAuthentication` method call from the `Configure` method.
+- Invoke the `AddJwtBearer` method in the `ConfigureServices` method:
+
+    ```csharp
+    services.AddJwtBearer(o => {
+        o.Audience = "http://localhost:5001/";
+        o.Authority = "http://localhost:5000/";
+    });
+    ```
+
+### Facebook Authentication
+Make the following changes in *Startup.cs*:
+
+- Remove the `UseFacebookAuthentication` method call from the `Configure` method.
+- Invoke the `AddFacebook` method in the `ConfigureServices` method:
+    
+    ```csharp
+    services.AddFacebook(o => {
+        o.AppId = Configuration["facebook:appid"];
+        o.AppSecret = Configuration["facebook:appsecret"];
+    });
+    ```
+
+### Google Authentication
+Make the following changes in *Startup.cs*:
+
+- Remove the `UseGoogleAuthentication` method call from the `Configure` method.
+- Invoke the `AddGoogle` method in the `ConfigureServices` method:
+
+    ```csharp
+    services.AddGoogle(o => {
+        o.ClientId = Configuration["auth:google:clientid"];
+        o.ClientSecret = Configuration["auth:google:clientsecret"];
+    });    
+    ```
+
+### Microsoft Account Authentication
+Make the following changes in *Startup.cs*:
+
+- Remove the `UseMicrosoftAccountAuthentication` method call from the `Configure` method.
+- Invoke the `AddMicrosoftAccount` method in the `ConfigureServices` method:
+
+    ```csharp
+    services.AddMicrosoftAccount(o => {
+        o.ClientId = Configuration["auth:microsoft:clientid"];
+        o.ClientSecret = Configuration["auth:microsoft:clientsecret"];
+    });
+    ``` 
+
+### Twitter Authentication
+Make the following changes in *Startup.cs*:
+
+- Remove the `UseTwitterAuthentication` method call from the `Configure` method.
+- Invoke the `AddTwitter` method in the `ConfigureServices` method:
+
+    ```csharp
+    services.AddTwitter(o => {
+        o.ConsumerKey = Configuration["auth:twitter:consumerkey"];
+        o.ConsumerSecret = Configuration["auth:twitter:consumersecret"];
+    });
+    ```
+
+### Multiple Authentication Schemes
+If your 1.x application has configured multiple authentication schemes, the default scheme needs to be identified differently in 2.0. 
+
+In 1.x, the `AutomaticAuthenticate` and `AutomaticChallenge` properties were intended to be set on a single authentication scheme. There was no good way to enforce this.
+
+In 2.0, these two properties have been removed as flags on the individual `AuthenticationOptions` instance and have moved into the base [AuthenticationOptions](/api/microsoft.aspnetcore.builder.authenticationoptions) class. The properties can be configured in the `AddAuthentication` method call within the `ConfigureServices` method of *Startup.cs*:
+
+```csharp
+services.AddAuthentication("Cookies");
```

Comment:
Use the CookieAuthenticationDefaults.AuthenticationScheme constant.

Explain that this set DefaultScheme.